content/working_with/manager/share-blueprint.md

```diff
@@ -0,0 +1,154 @@
+and will allow a better way of consuming shareable building blocks.      
+
+# Example
+
+In an Openstack cloud environment, a very basic building block could be a blueprint which creates a VM on open stack.
+
+Basic Openstack VM blueprint with a single port opened:
+
+{{< highlight  yaml >}}
+
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - http://www.getcloudify.org/spec/cloudify/4.5.5/types.yaml
+  - plugin:cloudify-openstack-plugin
+
+inputs:
+  port:
+    description: open port
+    default: 8080
+  agent_user:
+    description: User name used when SSH-ing into the started machine
+  image:
+    description: Openstack image name or id to use for the new server
+  flavor:
+    description: Openstack flavor name or id to use for the new server
+  network_name:
+    description: Openstack network name the new server will be connected to
+  floating_network_id:
+    description: The id of the network to use for allocating a floating ip
+  key_pair_name:
+    description: Openstack key pair name of the key to associate with the new server
+  private_key_path:
+    description: |
+      Path to the private key which will be used for connecting to the server
+      on the manager or machine running CLI if running in local mode.
+
+node_templates:
+  virtual_ip:
+    type: cloudify.openstack.nodes.FloatingIP
+    properties:
+      floatingip:
+        floating_network_id: { get_input: floating_network_id }
+
+  security_group:
+    type: cloudify.openstack.nodes.SecurityGroup
+    properties:
+      rules:
+        - port: { get_input: port }
+          remote_ip_prefix: 0.0.0.0/0
+        - port: 22
+          remote_ip_prefix: 0.0.0.0/0
+
+  keypair:
+    type: cloudify.openstack.nodes.KeyPair
+    properties:
+      use_external_resource: true
+      resource_id: { get_input: key_pair_name }
+      private_key_path: { get_input: private_key_path }
+
+  vm:
+    type: cloudify.openstack.nodes.Server
+    properties:
+      agent_config:
+        user: { get_input: agent_user }
+        key: { get_property: [ keypair, private_key_path ] }
+      image: { get_input: image }
+      flavor: { get_input: flavor }
+      management_network_name: { get_input: network_name }
+    relationships:
+      - type: cloudify.openstack.server_connected_to_keypair
+        target: keypair
+      - type: cloudify.openstack.server_connected_to_floating_ip
+        target: virtual_ip
+      - type: cloudify.openstack.server_connected_to_security_group
+        target: security_group
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          inputs:
+            args:
+              security_groups: [{ get_attribute: [ security_group, external_name ]}]
+{{< /highlight >}}
+
+
+In result, an application deployed on Openstack blueprint:
+
+{{< highlight  yaml >}}
+
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - openstack_infra->blueprint:openstack_vm
```

Comment:
Why not import twice, to show namespace collisions?

Reply:
there is one in the final blueprint example